app/services/forest_liana/has_many_getter.rb

```diff
@@ -4,26 +4,61 @@ def initialize(resource, association, params)
       @resource = resource
       @association = association
       @params = params
+      @field_names_requested = field_names_requested
+    end
+
+    def field_names_requested
+      return nil unless @params[:fields] && @params[:fields][@association.table_name]
+      @params[:fields][@association.table_name].split(',')
+                                            .map { |name| name.to_sym }
     end
 
     def perform
       @records = @resource
         .unscoped
         .find(@params[:id])
         .send(@params[:association_name])
+        .select(select)
```

Comment:
Yeah, the `unscoped` force the "select *" but I think it's not a big issue. The performance issue was due to the `joins`, not the `select`. Are you ok with that?